projects/plugins/jetpack/modules/sso.php

```diff
@@ -81,6 +82,116 @@ private function __construct() {
 			require_once JETPACK__PLUGIN_DIR . 'modules/sso/class-jetpack-force-2fa.php';
 			new Jetpack_Force_2FA();
 		}
+		add_filter( 'manage_users_columns', array( $this, 'jetpack_user_connected_th' ) );
+		add_action( 'admin_print_styles', array( $this, 'jetpack_user_table_styles' ) );
+		add_action( 'manage_users_custom_column', array( $this, 'jetpack_show_connection_status' ), 10, 3 );
+		add_action( 'admin_post_jetpack_invite_user_to_wpcom', array( $this, 'invite_user_to_wpcom' ) );
+		add_action( 'admin_notices', array( $this, 'handle_invitation_results' ) );
+	}
+
+	/**
+	 * Renders invitations errors/success messages in users.php.
+	 */
+	public function handle_invitation_results() {
+		$valid_nonce = isset( $_GET['_wpnonce'] ) ? wp_verify_nonce( $_GET['_wpnonce'], 'jetpack-sso-invite-user' ) : false; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput -- WP core doesn't pre-sanitize nonces either.
+
+		if ( ! isset( $_GET['jetpack-sso-invite-user'] ) || ! $valid_nonce ) {
+			return;
+		}
+		if ( $_GET['jetpack-sso-invite-user'] === 'success' ) {
+			return wp_admin_notice( __( 'Hooray! User was invited successfully!', 'jetpack' ), array( 'type' => 'success' ) );
+		}
+		if ( $_GET['jetpack-sso-invite-user'] === 'failed' && isset( $_GET['jetpack-sso-invite-error'] ) ) {
+			switch ( $_GET['jetpack-sso-invite-error'] ) {
+				case 'invalid-user':
+					return wp_admin_notice( __( 'Tried to invite a user that either doesn&#8217;t exist or doesn&#8217;t have an email address.', 'jetpack' ), array( 'type' => 'error' ) );
+				default:
+					// render whatever error we got from the API. It's escaped and nonce-checked.
+					return wp_admin_notice( esc_html( sanitize_text_field( wp_unslash( $_GET['jetpack-sso-invite-error'] ) ) ), array( 'type' => 'error' ) );
```

Comment:
If instead of a the error message we'd display a generic error, we could remove the nonce business here and in `invite_user_to_wpcom()`. Just something to consider.